artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/PostOfficeImpl.java

```diff
@@ -1306,7 +1306,7 @@ Bindings simpleRoute(SimpleString address, RoutingContext context, Message messa
       }
       if (bindings != null) {
          bindings.route(message, context);
-         if (addressInfo != null) {
+         if (addressInfo != null && context.getQueueCount() > 0) {
```

Comment:
Would it makes sense for simpleRoute to return null in this case, actually holding to its previously-implied contract, rather than return the non-null bindings and then requiring the caller to also know to check the Context (your other change above) like this does?